### PR TITLE
Add RSA Support

### PR DESCRIPTION
**Pull request by fmonniot**
Hello there,

I have added the RSA part of top of the `RSA` branch of your fork, which brought a few changes with it. I hope it's ok, otherwise I can rebase (I was trying to get your commit intact).

One thing I'm not super confident with this PR is the constructor of `OneKey`. Because the original was really tied to EC keys and to not change it too much (as well as the `ASN1` class), I created a second constructor which take `RSAPublicKey`/`RSAPrivateKey`. I ended up doing the same thing for `ASN1. DecodePKCS8` for the exact same reason.
I'm open to suggestion as to how we could merge this constructor/decoder in the more generic one, if you want to do so of course.

Please let me know what you think of this PR.

### CODE REVIEW

File: src/test/java/COSE/RegressionTest.java [hunk 1]
```diff
@@ -51,6 +65,8 @@
             "Examples/sign-tests",
             "Examples/sign1-tests",
             "Examples/RFC8152",
+            "Examples/rsa-pss-examples",
+            // "Examples/X509",
```
Comment (jimsch) [inline src/test/java/COSE/RegressionTest.java]: Don't include the X.509 or anima tests here.  Which means that there is probably some other clean up in this file that can be done.

**Comment by fmonniot** [comment]
Thank you for the review. I have pushed some changes which should resolve the requests you had. I tried to make a commit per comment, hopefully that will make it easier for you to review them.

For the constructor issue, what I did was to separate the ASN.1 decode method in three: one for the structure, one for the EC octetstring and the last for the RSA octetstring.

**Comment by jimsch** [comment]
No requirement - are you planning to do the RSA OAEP changes or not

**Comment by fmonniot** [comment]
Not really, I'm mainly interested in the signature part of the spec.

**Comment by fmonniot** [comment]
Hello @jimsch, do you have an idea of when you'll have time to take a look at the changes ? Thanks!

**Comment by fmonniot** [comment]
Thank you !

**Pull request merged**